### PR TITLE
Doc upgrades

### LINKED ISSUE #2227: Math not rendering in docs
Looks like https://docs.arbor-sim.org/en/latest/concepts/lif_cell.html#lifcell isn't rendering the math

### LINKED ISSUE #2228: Signature of spike structure is lacking some explanation
https://docs.arbor-sim.org/en/latest/python/simulation.html#arbor.simulation.spikes is missing the closing parenthesis, and it's not clearly explained what 'source', 'gid', 'index', or 'time' are.

Edit: not actually missing a parenthesis

### PR DESCRIPTION
**Pull request by thorstenhater**
- Polish & modernise two cells + gap junction example.
- Investigate problems with mathjax
- Update spike data format docs.
- Switches theme to FURO, which is better maintained and document (like ... at all)
  - with the new theme, we gain dark mode, yet we need to fix our graphics to make it a really nice experience.

Fixes #2227 
Fixes #2228

### CODE REVIEW

**Comment by Helveg** [comment]
How can I build the docs locally? Also see https://github.com/nest/nest-simulator/issues/2949 for a cool feature that makes docfixes much more accessible to first time contributors!

File: doc/python/simulation.rst [hunk 1]
```diff
@@ -112,11 +112,26 @@ over the local and distributed hardware resources (see :ref:`pydomdec`). Then, t
...
+            for (gid, index), time in sim.spikes():
+                pass
+
+        The fields ``gid`` and ``index`` identify the spike's source by the cell
+        global index ``gid`` and by the source index on that cell ``index``, eg
+        the ``n``th spike detector on a cable cell, and ``time`` is simply the
+        spike time.
+
+        The spikes are sorted in ascending order of spike time, and spikes with
+        the same time are sorted accourding to source gid then index.
```
Comment (AdhocMan) [inline doc/python/simulation.rst:134]:
```suggestion
        the same time are sorted according to source gid then index.
```

**Review by AdhocMan — COMMENTED** [review]
Looks great!
As you mentioned, some images still need work. 
These are the ones, where I've found issues with the dark theme:

- https://arbor--2229.org.readthedocs.build/en/2229/tutorial/network_two_cells_gap_junctions.html#
- https://arbor--2229.org.readthedocs.build/en/2229/ecosystem/index.html#simulators
- https://arbor--2229.org.readthedocs.build/en/2229/concepts/recipe.html#an-example-model
- https://arbor--2229.org.readthedocs.build/en/2229/concepts/hardware.html#hardware-context
- https://arbor--2229.org.readthedocs.build/en/2229/concepts/domdec.html#domain-decomposition

For the html based figures, it should be enough to change the text color to always be black. Maybe through some styling option specific to each div class?
The SVG figures could be fixed by setting the background to always be white. There appear to be ways to enforce this directly inside the SVG file.

**Review by AdhocMan — APPROVED** [review]
Looks great now!

**Pull request merged**